src/main/java/ca/teamdman/sfm/common/capabilityprovidermapper/ae2/InterfaceCapabilityProviderMapper.java

```diff
@@ -0,0 +1,446 @@
+                return LazyOptional.empty();
+            }
+
+            return in.getCapability(Capabilities.STORAGE_MONITORABLE_ACCESSOR);
+        }
+
+        <T> T withStorage(Function<MEStorage, T> callback) {
+            var cap = this.getCapability();
+            if (cap.isPresent()) {
+                //noinspection DataFlowIssue
+                return callback.apply(cap.map(c -> c.getInventory(IActionSource.empty())).orElse(null));
+            }
+            return null;
+        }
+
+        <T> T withBaseItemHandler(Function<IItemHandler, T> callback) {
+            var in = this.getInterface();
+            if (in == null) {
+                return null;
+            }
+
+            var maybeCap = in.getCapability(ForgeCapabilities.ITEM_HANDLER);
+            if (maybeCap.isPresent()) {
+                //noinspection DataFlowIssue
+                return callback.apply(maybeCap.orElse(null));
+            }
+            return null;
+        }
+
+        <T> T withBaseFluidHandler(Function<IFluidHandler, T> callback) {
+            var in = this.getInterface();
+            if (in == null) {
+                return null;
+            }
+
+            var maybeCap = in.getCapability(ForgeCapabilities.FLUID_HANDLER);
+            if (maybeCap.isPresent()) {
+                //noinspection DataFlowIssue
+                return callback.apply(maybeCap.orElse(null));
+            }
+            return null;
+        }
+
+        @Override
+        public int getSlots() {
+            Integer slots = this.withStorage(s -> {
+                int i = 0;
+                for (var stored : s.getAvailableStacks()) {
+                    if (stored.getKey() instanceof AEItemKey) {
+                        i++;
+                    }
+                }
+                return i;
+            });
+
+            if (slots == null) {
+                slots = this.withBaseItemHandler(IItemHandler::getSlots);
+            }
+
+            return slots == null ? 0 : slots;
+        }
+
+        @Override
+        public @NotNull ItemStack getStackInSlot(int slot) {
+            var stack = this.withStorage(s -> {
+                int i = 0;
+                for (var stored : s.getAvailableStacks()) {
+                    if (stored.getKey() instanceof AEItemKey key) {
+                        if (slot == i++) {
+                            return key.toStack((int) Math.min(Integer.MAX_VALUE, stored.getLongValue()));
+                        }
+                    }
+                }
+                return ItemStack.EMPTY;
+            });
+
+            if (stack == null) {
+                stack = this.withBaseItemHandler(c -> c.getStackInSlot(slot));
+            }
+
+            return stack == null ? ItemStack.EMPTY : stack;
+        }
+
+        @Override
+        public @NotNull ItemStack insertItem(int slot, @NotNull ItemStack stack, boolean simulate) {
```

Comment:
```suggestion
        public ItemStack insertItem(int slot, @NotNull ItemStack stack, boolean simulate) {
```
the package file should be adding notnullbydefault to everything